### ISSUE TITLE
Spotify Stops playing audio but keeps running

### ISSUE THREAD

**Issue by Awsome1uk** [marcelveldt/plugin.audio.spotify#78]
I am running Spotify 1.1.55 of Kodi 17.6 Gita9a7a20 running LibreELEC (official) 8.4.2 on RPI2.arm (Raspberry Pi 3). Spotify works correctly and I can navigate music without issues. However, when I play something it intermittently just stops playing the audio but the time code for track keeps increasing. The only way to rectify this is to stop Spotify and relaunch but it always comes back after a while. Other music applicatios work without issue. I have attached the kodi.log file for you to investigate. Hope you can identify the bug as I love this app in Kodi. Never had this issue with previous versios.

Thanks
Nigel
[01_KODI.log](https://github.com/marcelveldt/plugin.audio.spotify/files/1909927/01_KODI.log)

**Comment by gpchampion** [comment]
As per the issue I mentioned on OSMC  in another post using 1.1.55. I've reverted to 1.1.46 and it seems fine on local play back.

**Comment by chrisgo67** [comment]
I see a similiar behaviour with libreelec on my pi 3. Milhouse nightly 429.

**Comment by Martner44** [comment]
It still happens from time to time with my current set up as well (libreelec 8.2.3.1, kodi 17.6 on an S905x box). However, it has improved a lot after increasing <memorysize> & <readfactor> in advancedsettings.xml.

**Comment by Martner44** [comment]
Increasing memorysize & readfactor that is... 😉

**Comment by drrak** [comment]
The same problem.
Spotify 1.1.56, LibreELEC 8.2.5, RPi3

**Comment by fportegies** [comment]
Same here on 1.1.56 on libreelec on raspberry pi3. Music stops with short tick.
Lots of errors in the log. Sometimes sound is back after skip to new song, sometimes I have to reboot kodi. Logfile has lots of errors. https://pastebin.com/70peBLxD

**Comment by fportegies** [comment]
Hi Marcel, thank you for this great addon.
Sometimes the music stops playing due to connection problems with the Spotify website, quite annoying.
Logfile has CCurlFile-FillBuffer-Failed-Timeout-was-reached-28/

Something between the add-on and the Spotify website.
Greetings,
Frank

**Comment by drrak** [comment]
_Translator:_
Anyone know where I can get an older, better working version?

**Comment by Martner44** [comment]
Right 'round the corner... 😉

https://github.com/marcelveldt/repository.marcelveldt/find/master?q=

**Comment by fportegies** [comment]
Problem solved for me with new 3A power supply.

**Comment by Martner44** [comment]
Hey,
AFAIK, buffermode 0 will try to write to your (disk?)drive instead of RAM. Not knowing the weteks specs, this might be a little slow.

After a while I found out that that the trick with readmode & memorysize that improved the situation for me was actually reducing memorysize (5 MB) while only slightly increasing the readfactor (6.5). Honestly, I'm still not even sure if the change in readfactor really had an impact on the issue, at all. It felt like it. 

However, changing those values will probably only have a positive effect on fast machines with gigabit connections.

The logic behind this workaround is, based on the assumption that the root cause is a buffer underrun, make the buffer smaller and load it faster. Again, honestly, I have absolutely no proof so far that my assumption is correct. For some reason, I do get a lot less drop outs or 'ghost tracks' since I made these changes. About 1 every 1,5 hours, roughly.

**Comment by maxenced** [comment]
Hmm, isn't buffermode only setting what is cached and  not how ? https://kodi.wiki/view/HOW-TO:Modify_the_video_cache#Cache_settings 

I will give your setting a try anyway (reducing memorysize at first).

Thanks for helping, will send update in a few days

**Comment by gpchampion** [comment]
Indeed, I'm back at 1.1.46 and its a little more reliable, less ghost tracks....

**Comment by chrisgo67** [comment]
Sorry to say but i still have no sound with 1.1.44 or 1.1.46 on my Pi 3.
Does anybody have an idea what i could do to get thecsound back?
All orher addons and systemsounds work fine.

**Comment by Martner44** [comment]
No sound at all?

I'm not sure this is related to this specific add-on.

Can you play music files from a local disk or network drive?

**Comment by chrisgo67** [comment]
Yes, all other afdons an players play fine. The sound is only misding in the spotify addon.

**Comment by fportegies** [comment]
I had problems like this on the pi3, caused by a bad power supply (2A). Now with a new (2,5A) power supply everything works fine also spotify addon no errors.

**Comment by chrisgo67** [comment]
Aftet upgrading to the latest nighly milhouse bild #517 yesterday everything works fine with version .44.

**Comment by maxenced** [comment]
@Martner44 looks like you were right, reducing the cache helped a lot ! 
I set it to 5M, readfactor of 4, and now have issue only ~ 1 time every hour or so, instead of every 2/3 songs !

**Comment by gpchampion** [comment]
Is the advanced settings changes with version 1.1.44 or the latest at 1.1.56? Plus, any idea why it helps and does it affect anything else? Thanks.

**Comment by maxenced** [comment]
I don't think it changed as these settings are kodi related, not local to this plugin. We (with @Martner44 ) guess that under some condition, this plugin fails to play with cache buffer and wait for it to be filled. With a smaller cache size, the time to wait may be smaller. We don't have a strong explanation though ...

I use kodi to watch lot of video that are nfs-mounted (series / movies) and didn't see any problem with this new setting.

**Comment by Martner44** [comment]
@maxenced: Glad my workaround works for you, as well 👍

Regarding the versions, I've downgraded step by step from 1.1.56 to 1.1.44 in the meantime. I haven't really noticed any different behaviour, to be honest... On the other hand, I didn't spend hours on testing each version, so... who knows.

**Comment by Martner44** [comment]
@drrak: As I didn't notice any difference yet, I might as well stick with 1.1.44.

**Comment by drrak** [comment]
_Translator_:
Btw. I'm using the original/unchanged advancedsettings.xml file.

**Comment by Martner44** [comment]
@ddrak: what hardware/machine are you using?

**Comment by drrak** [comment]
RPi3

**Comment by Martner44** [comment]
OK, I'm giving it a try, without advancedsettings.xml, tomorrow. Let's see what happens.

**Comment by Martner44** [comment]
So, after 3 hours of playing music using v. 1.1.44 on my Nexbox A95X (CoreELEC 8.90.2) with a) increased buffer (10 MB & readfactor 6.5) and another 1,5 hours with b) advancedsettings.xml turned off (i.e. ~20 MB standard buffer & readfactor 4.0), I got 1 dropout/ghost track under a) and no single one under b). 

So far, so good. 

However, with both settings, I keep on getting the usual 

ERROR:CCurlFile::FillBuffer - Failed: Transferred a partial file(18) 

in the log with each and every track/file.

After turning advancedsettings.xml on again & reducing the buffer to 8 MB (--> <memorysize>8388608</memorysize>), I still get no dropouts/ghost tracks and the CCurlFile Error only every 2 - 3 tracks. Readfactor remained at 6.5.

I am actually willing to see that as an improvement... ;-) Mostly, because I'm growing tired to mess with it. And as long as the errors in the log don't lead to audible disturbances... who cares...?

Cheers, over & out.

**Comment by drrak** [comment]
_Translator_:
Unfortunately, version 44 has a problem with long songs. It was already mentioned in this thread: #55
